confgenerator/confgenerator.go

```diff
@@ -383,12 +386,55 @@ func (l *Logging) generateFluentbitComponents(userAgent string, hostInfo *host.I
 		BufferInMemory: true,
 	}.Components("ops-agent-fluent-bit")...)
 
-	out = append(out, stackdriverOutputComponent("ops-agent-fluent-bit", userAgent))
+	out = append(out, generateSeveritySelfLogsParser()...)
+
+	out = append(out, stackdriverOutputComponent(fluentBitSelfLogTag, userAgent))
 	out = append(out, fluentbit.MetricsOutputComponent())
 
 	return out, nil
 }
 
+func generateSeveritySelfLogsParser() []fluentbit.Component {
+	out := make([]fluentbit.Component, 0)
+
+	_, timeOffset := time.Now().Zone()
+
+	parser := LoggingProcessorParseRegex{
+		Regex:       `(?<message>\[[ ]*(?<time>\d+\/\d+\/\d+ \d+:\d+:\d+)] \[[ ]*(?<severity>[a-z]+)\].*)`,
+		PreserveKey: true,
+		ParserShared: ParserShared{
+			TimeKey:    "time",
+			TimeFormat: "%Y/%m/%d %H:%M:%S",
+			TimeOffset: fmt.Sprintf("%d", timeOffset),
+			Types: map[string]string{
+				"severity": "string",
+			},
+		},
+	}.Components("logging-severity-parser", "self-logs-severity")
+
+	out = append(out, parser...)
+
+	out = append(out, fluentbit.Component{
+		Kind: "FILTER",
+		Config: map[string]string{
+			"Key_Name": "message",
+			"Name":     "parser",
+			"Parser":   "logging-severity-parser.self-logs-severity",
+			"Match":    fluentBitSelfLogTag,
+		},
+	})
+
```

Comment:
Question: Do we need the extra `FILTER` here? Can we let `LoggingProcessorParseRegex` match on `fluentBitSelfLogTag` directly? 
```suggestion
	}.Components(fluentBitSelfLogTag, "self-logs-severity")

	out = append(out, parser...)

```

Reply:
Removed extra filter, and adding integration tests